packages/providers/upload-aws-s3/lib/index.js

```diff
@@ -15,19 +15,34 @@ function assertUrlProtocol(url) {
 }
 
 module.exports = {
-  init(config) {
+  init({ baseUrl = null, rootPath = null, s3Options, ...legacyS3Options }) {
+    if (legacyS3Options) {
```

Comment:
`legacyS3Options` will always be a truthy value because is an object, you will need to check how many keys it has

```js
    if (Object.keys(legacyS3Options).length) {
```

Reply:
https://github.com/strapi/strapi/pull/16327